spring-hibernate4/src/test/java/com/baeldung/hibernate/oneToMany/main/HibernateOneToManyAnnotationMainTest.java

```diff
@@ -11,30 +20,56 @@
 
 public class HibernateOneToManyAnnotationMainTest {
     
+    private static SessionFactory sessionFactory;
+
+    private Session session;
+    
     public HibernateOneToManyAnnotationMainTest() {
     }
     
-    @BeforeClass
-    public static void setUpClass() {
-    }
     
-    @AfterClass
-    public static void tearDownClass() {
+    @BeforeClass
+    public static void beforeTests() {
+        Configuration configuration = new Configuration().addAnnotatedClass(Cart.class).addAnnotatedClass(Items.class).setProperty("hibernate.dialect", HSQLDialect.class.getName()).setProperty("hibernate.connection.driver_class", org.hsqldb.jdbcDriver.class.getName())
+                .setProperty("hibernate.connection.url", "jdbc:hsqldb:mem:test").setProperty("hibernate.connection.username", "sa").setProperty("hibernate.connection.password", "").setProperty("hibernate.hbm2ddl.auto", "update");
+        ServiceRegistry serviceRegistry = new StandardServiceRegistryBuilder().applySettings(configuration.getProperties()).build();
+        sessionFactory = configuration.buildSessionFactory(serviceRegistry);
     }
     
     @Before
     public void setUp() {
+        session = sessionFactory.openSession();
+        session.beginTransaction();  
+        
     }
     
-    @After
-    public void tearDown() {
+    
+
+    @Test
+    public void testAddItemsToCart() {
+        Cart cart = new Cart();
+        Items item1 = new Items("I10", 10, 1, cart);
+        Set<Items> itemsSet = new HashSet<Items>();
+        assertEquals(0, itemsSet.size());
```

Comment:
This is standard Java feature. It should not be a part of the test, the same for checking if the item1 object is null